js/style-spec/API.md

```diff
@@ -0,0 +1,227 @@
+
+If `space` is unspecified, a default of `2` spaces will be used.
+
+
+### Parameters
+
+| parameter | type   | description                                            |
+| --------- | ------ | ------------------------------------------------------ |
+| `style`   | Object | a Mapbox GL Style                                      |
+| `[space]` | number | _optional:_ space argument to pass to `JSON.stringify` |
+
+
+### Example
+
+```js
+var fs = require('fs');
+var format = require('mapbox-gl-style-spec').format;
+var style = fs.readFileSync('./source.json', 'utf8');
+fs.writeFileSync('./dest.json', format(style));
+fs.writeFileSync('./dest.min.json', format(style, 0));
+```
+
+
+**Returns** `string`, stringified formatted JSON
+
+
+### `groupByLayout(layers)`
+
+Given an array of layers, return an array of arrays of layers where all
+layers in each group have identical layout-affecting properties. These
+are the properties that were formerly used by explicit `ref` mechanism
+for layers: 'type', 'source', 'source-layer', 'minzoom', 'maxzoom',
+'filter', and 'layout'.
+
+The input is not modified. The output layers are references to the
+input layers.
+
+
+### Parameters
+
+| parameter | type             | description |
+| --------- | ---------------- | ----------- |
+| `layers`  | Array\.\<Layer\> |             |
+
+
+
+**Returns** `Array.<Array.<Layer>>`, 
+
+
+### `migrate(style)`
+
+Migrate a Mapbox GL Style to the latest version.
+
+
+### Parameters
+
+| parameter | type   | description       |
+| --------- | ------ | ----------------- |
+| `style`   | object | a Mapbox GL Style |
+
+
+### Example
+
+```js
+var fs = require('fs');
+var migrate = require('mapbox-gl-style-spec').migrate;
+var style = fs.readFileSync('./style.json', 'utf8');
+fs.writeFileSync('./style.json', JSON.stringify(migrate(style)));
+```
+
+
+**Returns** `Object`, a migrated style
+
+
+### `validate(style, [styleSpec])`
+
+Validate a Mapbox GL style against the style specification.
+
+
+### Parameters
+
+| parameter     | type                   | description                                                                                                                |
+| ------------- | ---------------------- | -------------------------------------------------------------------------------------------------------------------------- |
+| `style`       | Object\,String\,Buffer | The style to be validated. If a `String`     or `Buffer` is provided, the returned errors will contain line numbers.       |
+| `[styleSpec]` | Object                 | _optional:_ The style specification to validate against.     If omitted, the spec version is inferred from the stylesheet. |
+
+
+### Example
+
+```js
+  var validate = require('mapbox-gl-style-spec').validate;
+  var style = fs.readFileSync('./style.json', 'utf8');
+  var errors = validate(style);
+```
+
+
+**Returns** `Array.<ValidationError|ParsingError>`, 
+
+
+### `validateStyleMin(style, [styleSpec])`
```

Comment:
Custom browserify approach means we can merge this with `validate`. 👍  if you want to do that in a separate PR.